src/app/games/hive/HiveMove.ts

```diff
@@ -0,0 +1,170 @@
+        (fields: [HivePiece, Coord]): HiveMoveDrop => new HiveMoveDrop(fields[0], fields[1].x, fields[1].y),
+    );
+
+    public static from(piece: HivePiece, coord: Coord): HiveMoveDrop {
+        return new HiveMoveDrop(piece, coord.x, coord.y);
+    }
+
+    private constructor(public readonly piece: HivePiece, x: number, y: number) {
+        super(x, y);
+    }
+
+    public toString(): string {
+        return `HiveDrop(${this.piece.toString()}, ${this.coord.toString()})`;
+    }
+
+    public equals(other: HiveMove): boolean {
+        if (other instanceof HiveMoveDrop) {
+            return this.piece.equals(other.piece) && this.coord.equals(other.coord);
+        }
+        return false;
+    }
+}
+
+export class HiveMoveCoordToCoord extends MoveCoordToCoord {
+
+    public static encoder: Encoder<HiveMoveCoordToCoord> = MoveCoordToCoord.getEncoder((start: Coord, end: Coord) => {
+        return new HiveMoveCoordToCoord(start, end);
+    });
+
+    public static from(start: Coord, end: Coord): MGPFallible<HiveMoveCoordToCoord> {
+        if (start.equals(end)) {
+            return MGPFallible.failure(RulesFailure.MOVE_CANNOT_BE_STATIC());
+        }
+        return MGPFallible.success(new HiveMoveCoordToCoord(start, end));
+    }
+
+    protected constructor(start: Coord, end: Coord) {
+        super(start, end);
+    }
+
+    public toString(): string {
+        return `HiveMoveCoordToCoord(${this.coord.toString()} -> ${this.end.toString()})`;
+    }
+
+    public equals(other: HiveMove): boolean {
+        if (other instanceof HiveMoveSpider) {
+            // Spider moves are coord to coord but the intermediate coords matter, so we can't compare them here
+            return false;
+        }
+        if (other instanceof HiveMoveCoordToCoord) {
+            return this.coord.equals(other.coord) && this.end.equals(other.end);
+        }
+        return false;
+    }
+}
+
+export class HiveMoveSpider extends HiveMoveCoordToCoord {
+
+    public static encoder: Encoder<HiveMoveSpider> = MoveEncoder.tuple(
+        [Coord.encoder, Coord.encoder, Coord.encoder, Coord.encoder],
+        (move: HiveMoveSpider): [Coord, Coord, Coord, Coord] => move.coords,
+        (fields: [Coord, Coord, Coord, Coord]): HiveMoveSpider => new HiveMoveSpider(fields),
+    );
+
+    public static fromCoords(coords: [Coord, Coord, Coord, Coord]): HiveMoveSpider {
+        return new HiveMoveSpider(coords);
+    }
+
+    private constructor(public readonly coords: [Coord, Coord, Coord, Coord]) {
+        super(coords[0], coords[3]);
+    }
+
+    public toString(): string {
+        const coords: string = this.coords.map((coord: Coord) => coord.toString()).join(', ');
+        return `HiveMoveSpider(${coords})`;
+    }
+
+    public equals(other: HiveMove): boolean {
+        if (other instanceof HiveMoveSpider) {
+            return this.coord.equals(other.coord) && this.end.equals(other.end);
```

Comment:
Nan mais attiiin.... Tu te soucie quand même que de start et end in fine oO ?
J'm'enconfusoie

Reply:
oui !

Reply:
Donc le equals considère tout les A**D égaux mais on stock bien B et C pour que les deux joueurs voient bien la même chose, c'bien ça ?

Reply:
yes!